File: z.Abandoned/Deprecated/script_generateCommandJson/generateCommandJson.py
import os
import json
import zipfile
from pathlib import Path

def normalizePath(inputPath):
    return inputPath.strip().replace('"', '').replace("'", '')

inputPath = input('Drag the "data.json" file into this terminal: ')
inputPath = normalizePath(inputPath)

if Path(inputPath).name != "data.json":
    print('File isn\'t "data.json"!')
    exit()

projectDir = os.path.dirname(inputPath)
rawData = Path(inputPath).read_text(encoding="utf-8")
botData = json.loads(rawData)
commandsFolder = os.path.join(projectDir, botData["name"])

if not os.path.exists(commandsFolder):
    os.makedirs(commandsFolder, exist_ok=True)

commands = botData["commands"]
for command in commands:
    commandFilePath = os.path.join(commandsFolder, f"{command['name']}.json")
    with open(commandFilePath, "w", encoding="utf-8") as f:
        json.dump(command, f, indent=2)
    print(f"{commandFilePath} created...")

print(f"Converted All Commands Into Individual Json In {commandsFolder}")

answer = input("\nCompress To Zip? (y/n): ")
if answer.strip().lower() == "y":
    zipFile = f"{commandsFolder}.zip"
    print("Creating Zip...")
    with zipfile.ZipFile(zipFile, "w", zipfile.ZIP_DEFLATED) as zipf:
        for root, _, files in os.walk(commandsFolder):
            for file in files:
                filePath = os.path.join(root, file)
                arcname = os.path.relpath(filePath, os.path.dirname(commandsFolder))
                zipf.write(filePath, arcname)
    print(f"Zip Creation Success: {zipFile}")
else:
    print("Skipping Zip Creation...")

File: z.Abandoned/Deprecated/script_generateCommandJson/generateCommandJson.js
const fs = require("node:fs")
const path = require("node:path")
const readline = require("node:readline")
const { exec } = require("node:child_process")
const { stdout, stderr } = require("node:process")
const { error } = require("node:console")

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
})

rl.question(`Drag the "data.json" file into this terminal: `, (inputPath)=>{
  inputPath = inputPath.replaceAll(`"`, "")

  if (path.basename(inputPath) !== "data.json"){
    console.error(`File isn't "data.json"!`)
    rl.close()
    return
  }

  let projectDir = path.dirname(inputPath)
  let rawData = fs.readFileSync(path.normalize(inputPath), "utf-8")
  let botData = JSON.parse(rawData)
  const commandsFolder = path.join(projectDir, botData.name)

  if(!fs.existsSync(commandsFolder)){
    fs.mkdirSync(commandsFolder, {recursive: true})
  }

  const commands = botData.commands
  commands.forEach(command =>{
    let commandFilePath = path.join(commandsFolder, `${command.name}.json`)
    fs.writeFileSync(commandFilePath, JSON.stringify(command, null, 2), "utf-8")
    console.log(`${commandFilePath} created...`)
  })

  console.log(`Converted All Commands Into Individual Json In ${commandsFolder}`)

  rl.question(`\nCompress To Zip? (y/n): `, (answer)=>{
    if(answer.trim().toLowerCase() == "y"){
      const zipFile = `${commandsFolder}.zip`
      const zipCommand = process.platform === 'win32'
        ? `powershell -command "Compress-Archive -Path '${commandsFolder}\\*' -DestinationPath '${zipFile}'"`
        : `zip -r "${zipFile}" "${path.basename(commandsFolder)}"`

      console.log(`Creating Zip...`)
      exec(zipCommand, {cwd: projectDir}, (error, stdout, stderr)=>{
        if (error){
          console.error(`Zip Creation Failed...`)
        } else {
          console.log(`Zip Creation Success: ${zipFile}`)
        }
        rl.close()
      })
    } else {
      console.log(`Skipping Zip Creation...`)
      rl.close()
    }
  })
})